src/main/java/org/myrobotlab/service/Gpt3.java

```diff
@@ -141,6 +167,22 @@ public Response getResponse(String text) {
     return null;
   }
   
+  /**
+   * Overriden error to also publish the errors
+   * probably would be a better solution to self subscibe to errors and 
+   * have the subscriptions publish utterances/responses/text
+   */
+  public Status error(String error) {
+    Status status = super.error(error);
+    invoke("publishText", error);    
+    Response response = new Response("friend", getName(), error, null);

```

Comment:
Why "friend" here?

Reply:
its a ProgramAB Response object ... until it gets promoted to something more generalized.
And the Response object expect to know the "user" the bot is chatting with ... default is friend